### PR TITLE
Add rotate cert api action

### PR DESCRIPTION
**Pull request by galal-hussein**
https://github.com/rancher/types/pull/622
https://github.com/rancher/rke/pull/1032
https://github.com/rancher/kontainer-engine/pull/97

https://github.com/rancher/rke/issues/450

### CODE REVIEW

File: pkg/api/customization/cluster/actionhandler.go [hunk 1]
```diff
@@ -215,6 +217,42 @@ func (a ActionHandler) ExportYamlHandler(actionName string, action *types.Action
...
+	if err = json.Unmarshal(data, &input); err != nil {
+		return errors.Wrap(err, "unmarshaling input error")
+	}
+
+	rotateCerts := &v3.RotateCertificates{
+		CACertificates: input.CACertificates,
+		Services:       []string{input.Services},
+	}
+	cluster.Spec.RancherKubernetesEngineConfig.RotateCertificates = rotateCerts
+	a.ClusterClient.Update(cluster)
```
Comment (alena1108) [inline pkg/api/customization/cluster/actionhandler.go]: we don't care if the update fails?

File: pkg/api/customization/cluster/actionhandler.go [hunk 2]
```diff
@@ -481,3 +483,38 @@ func (a ActionHandler) getKubeConfig(apiContext *types.APIContext, cluster *mgmt
 
 	return a.ClusterManager.KubeConfig(cluster.ID, token), nil
 }
+
+func (a ActionHandler) RotateCertificates(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	cluster, err := a.ClusterClient.Get(apiContext.ID, metav1.GetOptions{})
```
Comment (kinarashah) [inline pkg/api/customization/cluster/actionhandler.go]: I think we should be using access.ByID to access cluster object, 
https://github.com/rancher/rancher/blob/master/pkg/api/customization/cluster/actionhandler.go#L72

File: pkg/api/customization/cluster/actionhandler.go [hunk 3]
```diff
@@ -481,3 +483,38 @@ func (a ActionHandler) getKubeConfig(apiContext *types.APIContext, cluster *mgmt
...
+	if err = json.Unmarshal(data, &input); err != nil {
+		return errors.Wrap(err, "unmarshaling input error")
+	}
+
+	rotateCerts := &v3.RotateCertificates{
+		CACertificates: input.CACertificates,
+		Services:       []string{input.Services},
+	}
+	cluster.Spec.RancherKubernetesEngineConfig.RotateCertificates = rotateCerts
+	a.ClusterClient.Update(cluster)
```
Comment (kinarashah) [inline pkg/api/customization/cluster/actionhandler.go]: need to check for error?

File: pkg/api/customization/cluster/actionhandler.go (hunk 2 above)
Comment (galal-hussein) [reply pkg/api/customization/cluster/actionhandler.go]: will fix

**Pull request merged**